boa3_test/tests/boatestcase.py

```diff
@@ -0,0 +1,544 @@
+            _CONTRACT_LOCK.release()
+
+        super().tearDown()
+
+    @property
+    def contract(self) -> Optional[GenericContract]:
+        if self._contract is None and self.contract_hash is not None:
+            _CONTRACT_LOCK.acquire()
+            print(f"'Enter test method {self._testMethodName if hasattr(self, '_testMethodName') else self.id()}'")
+            self._contract = GenericContract(self.contract_hash)
+
+        return self._contract
+
+    @classmethod
+    async def set_up_contract(cls,
+                              contract_path: str,
+                              *,
+                              output_name: str = None,
+                              change_manifest_name: bool = False,
+                              signing_account: account.Account = None
+                              ) -> GenericContract:
+
+        cls.contract_hash = await cls.compile_and_deploy(contract_path,
+                                                         output_name=output_name,
+                                                         change_manifest_name=change_manifest_name,
+                                                         signing_account=signing_account
+                                                         )
+
+        return cls.contract
+
+    @classmethod
+    async def compile_and_deploy(cls,
+                                 contract_path: str,
+                                 *,
+                                 output_name: str = None,
+                                 change_manifest_name: bool = False,
+                                 signing_account: account.Account = None
+                                 ) -> types.UInt160:
+
+        nef_abs_path, _ = cls.get_deploy_file_paths(contract_path,
+                                                    output_name=output_name,
+                                                    change_manifest_name=change_manifest_name,
+                                                    )
+
+        if not signing_account:
+            signing_account = cls.genesis
+
+        return await cls.deploy(nef_abs_path, signing_account)
+
+    def unwrap_inner_values(self, value: list | dict):
+        if isinstance(value, list):
+            for index, item in enumerate(value):
+                if not isinstance(item, noderpc.StackItem):
+                    continue
+
+                value[index] = self._unwrap_stack_item(item)
+
+        elif isinstance(value, dict):
+            aux = value.copy()
+            value.clear()
+            for key, item in aux.items():
+                dict_key = key if not isinstance(key, noderpc.StackItem) else self._unwrap_stack_item(key)
+                dict_item = item if not isinstance(item, noderpc.StackItem) else self._unwrap_stack_item(item)
+
+                value[dict_key] = dict_item
+
+    def _unwrap_stack_item(self, stack_item: noderpc.StackItem) -> Any:
```

Comment:
These two methods were implemented to properly assert the result of lists and dicts. `call` method returns a list of StackItems for example instead of deserializing the inner objects, so I exposed the `unwrap_inner_values` to handle list and dict outputs.
https://github.com/CityOfZion/neo3-boa/blob/b2ed11822af1c90afb7507d5c138befe3a08ff62/boa3_test/tests/compiler_tests/test_arithmetic.py#L379-L404

boa-test-constructor doesn't have a method to unwrap the values, its logic is directly on `call` method, but I needed it to recursively handle the inner values on collection.
https://github.com/CityOfZion/boa-test-constructor/blob/d1c33d9cd2d6953326a79e008c01fb46e9a3690f/boaconstructor/__init__.py#L144-L166
I just didn't see the need to expose `unwrap_stack_item` since it shouldn't have StackItems returning from the test case classes, that's why this is set as private